sweeper.lua

```diff
@@ -0,0 +1,45 @@
+local S = moreblocks.intllib
+local check_protection = moreblocks.check_protection
+local sweepables = {}
+
+-- example sweepable node
+-- TODO API for sweepable node registration
+sweepables["default:mossycobble"] = function(pos, node)
+	-- hit 3 times to clean
+	if node.param1 < 2 then
+		node.param1 = node.param1 + 1
+		minetest.set_node(pos, node)
+		return false
+	end
+
+	-- set clean node
+	minetest.set_node(pos, { name = "default:cobble" })
```

Comment:
You should consider using `minetest.swap_node` here; it would keep the metadata of the node intact if you ever allowed registering nodes with metadata or inventories (mossy chest to normal chest or something like that).